main.go

```diff
@@ -469,6 +469,11 @@ func main() {
 		*s3prefix = *logURL
 	}
 
+	_, err := url.ParseRequestURI(*s3bucket)
+	if err == nil {
+		log.Fatal("scheme provided for s3-bucket, but should not be")
+	}
```

Comment:
Looking at this with a fresh set of eyes this morning I realize I've somewhat lead you astray. If your goal is to validate that input as an acceptable "bucket name" we should take a look at the [S3 docs](https://docs.aws.amazon.com/AmazonS3/latest/userguide/bucketnamingrules.html).

Looks like we can accomplish the job using `net.ParseIP()` and a small regular expression. This is by no means completely covering the requirements, but I think catching the most obvious of mistakes should be good enough.

```go
	if net.ParseIP(bucketName) != nil {
		return errors.New("-bucket-name cannot be an IP address"
	}

	// 1. 3-63 characters in length.
	// 2. Only contains lowercase letters, numbers, and hyphens.
	// 3. Must begin and end with a letter or number.
	pattern := `^[a-z0-9](?:[-a-z0-9]{1,61}[a-z0-9])?$`
...
```

Reply:
I think that work should be left up to the aws-sdk-go or whichever tool initially builds the s3 bucket. This tool is going to make an attempt to connect and will return an error if the bucket can't be reached.